pythonforandroid/recipe.py

```diff
@@ -1090,6 +1244,115 @@ def reduce_object_file_names(self, dirn):
+                join(recipe_build_dir,
+                     'config.guess'))().stdout.strip().decode('utf-8')
+
+            if not exists('config.status'):
+                shprint(
+                    sh.Command(join(recipe_build_dir, 'configure')),
+                    *(' '.join(self.configure_args).format(
+                                    android_host=env['HOSTARCH'],
+                                    android_build=android_build,
+                                    prefix=sys_prefix,
+                                    exec_prefix=sys_exec_prefix)).split(' '),
+                    _env=env)
+
+            if not exists('python'):
+                shprint(sh.make, 'all', _env=env)
+
+            # TODO: Look into passing the path to pyconfig.h in a
+            # better way, although this is probably acceptable
+            sh.cp('pyconfig.h', join(recipe_build_dir, 'Include'))
+
+
+class TargetHostPythonRecipe(Recipe):
+    '''
+    This is the base class for the hostpython recipes. This class will take
+    care to do all the work to build a hostpython recipe but, be careful, it
+    is intended to be subclassed because some of the vars needs to be set:
+
+        - :attr:`name`
+        - :attr:`version`
+
+    .. versionadded:: 0.6.0
+    '''
+
+    name = ''
+    '''The hostpython's recipe name. This should be ``hostpython2`` or 
+    ``hostpython3``
+
+    .. warning:: This must be set in inherited class.'''
+
+    version = ''
+    '''The hostpython's recipe version.
+
+    .. warning:: This must be set in inherited class.'''
+
+    build_subdir = 'native-build'
+    '''Specify the sub build directory for the hostpython recipe. Defaults
+    to ``native-build``.'''
+
+    url = 'https://www.python.org/ftp/python/{version}/Python-{version}.tgz'
+    '''The default url to download our host python recipe. This url will
+    change depending on the python version set in attribute :attr:`version`.'''
+
+    def get_build_container_dir(self, arch=None):
+        choices = self.check_recipe_choices()
+        dir_name = '-'.join([self.name] + choices)
+        return join(self.ctx.build_dir, 'other_builds', dir_name, 'desktop')
+
+    def get_build_dir(self, arch=None):
+        '''
+        .. note:: Unlike other recipes, the hostpython build dir doesn't
+            depend on the target arch
+        '''
+        return join(self.get_build_container_dir(), self.name)
+
+    def get_path_to_python(self):
+        return join(self.get_build_dir(), self.build_subdir)
+
+    def build_arch(self, arch):
+        recipe_build_dir = self.get_build_dir(arch.arch)
+
+        # Create a subdirectory to actually perform the build
+        build_dir = join(recipe_build_dir, self.build_subdir)
+        ensure_dir(build_dir)
+
+        if not exists(join(build_dir, 'python')):
+            with current_directory(recipe_build_dir):
+                # Configure the build
+                with current_directory(build_dir):
```

Comment:
I'm curious to know if we can we skip the `with current_directory(recipe_build_dir)` above and just use this one?
Unless or maybe we're dealing with relative directories.
If it's the case and both are still needed, maybe you want to give a try to:
```python
with current_directory(recipe_build_dir), current_directory(build_dir):
```
So you don't have to indent two levels.